src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx

```diff
@@ -1,5 +1,236 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { selectProviderSelectionInfo } from '../../../utils/selectors';
+import ResidentialAddress from '../../../components/ResidentialAddress';
+import { connect } from 'react-redux';
+import * as actions from '../../redux/actions';
+import { FETCH_STATUS } from '../../../utils/constants';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+import { distanceBetween } from '../../../utils/address';
+import { scrollAndFocus } from '../../../utils/scrollAndFocus';
+import ErrorMessage from '../../../components/ErrorMessage';
 
-export default function ProviderSelectionField() {
-  return <strong>List stuff goes here</strong>;
+const INITIAL_PROVIDER_DISPLAY_COUNT = 5;
+
+function ProviderSelectionField({
+  address,
+  formData,
+  onChange,
+  idSchema,
+  requestStatus,
+  requestProvidersList,
+  communityCareProviderList,
+}) {
+  const [checkedProvider, setCheckedProvider] = useState();
+  const [mounted, setMounted] = useState(false);
+  const [showProvidersList, setShowProvidersList] = useState(false);
+  const [providersListLength, setProvidersListLength] = useState(
+    INITIAL_PROVIDER_DISPLAY_COUNT,
+  );
+  const currentlyShownProvidersList = communityCareProviderList?.slice(
+    0,
+    providersListLength,
+  );
+  const loadingProviders =
+    requestStatus === FETCH_STATUS.loading ||
+    requestStatus === FETCH_STATUS.notStarted;
+  const providerSelected = 'id' in formData;
+  useEffect(() => {
+    requestProvidersList(address);
+    setMounted(true);
+  }, []);
+
+  useEffect(
+    () => {
+      if (showProvidersList && !loadingProviders) {
+        scrollAndFocus('h2');
+      } else if (mounted && !loadingProviders) {
```

Comment:
Should this be `!showProvidersList` instead of `!loadingProviders`? I don't think the loading state matters for setting the focus back to the button link.

Reply:
@jbalboni I had actually added both of those `!loadingProviders` in my shuffling around of the `requestProvidersList()`. We originally didn't have them in this useEffect. Do you still want it changed or removed as it doesn't matter?

Reply:
Yeah, it still makes sense to change it. The condition you have in the first `if` statement looks good, it's just the `else if` that I think needs to be tweaked.

Reply:
Fixed